Explorer/Assets/DCL/SDKComponents/SceneUI/Systems/UIInput/UIInputInstantiationSystem.cs

```diff
@@ -0,0 +1,60 @@
+﻿using Arch.Core;
+using Arch.System;
+using Arch.SystemGroups;
+using DCL.Diagnostics;
+using DCL.ECSComponents;
+using DCL.Optimization.Pools;
+using DCL.SDKComponents.SceneUI.Classes;
+using DCL.SDKComponents.SceneUI.Components;
+using DCL.SDKComponents.SceneUI.Groups;
+using DCL.SDKComponents.SceneUI.Utils;
+using ECS.Abstract;
+using UnityEngine.UIElements;
+
+namespace DCL.SDKComponents.SceneUI.Systems.UIInput
+{
+    [UpdateInGroup(typeof(SceneUIComponentInstantiationGroup))]
+    [LogCategory(ReportCategory.SCENE_UI)]
+    public partial class UIInputInstantiationSystem : BaseUnityLoopSystem
+    {
+        private readonly IComponentPool<TextField> textFieldsPool;
+
+        public UIInputInstantiationSystem(World world, IComponentPoolsRegistry poolsRegistry) : base(world)
+        {
+            textFieldsPool = poolsRegistry.GetReferenceTypePool<TextField>();
+        }
+
+        protected override void Update(float t)
+        {
+            InstantiateUIInputQuery(World);
+            UpdateUIInputQuery(World);
+        }
+
+        [Query]
+        [All(typeof(PBUiInput), typeof(PBUiTransform), typeof(UITransformComponent))]
+        [None(typeof(UIInputComponent))]
+        private void InstantiateUIInput(in Entity entity, ref UITransformComponent uiTransformComponent)
+        {
+            var textField = textFieldsPool.Get();
+            textField.name = $"UIInput (Entity {entity.Id})";
+            textField.AddToClassList("dcl-input");
+            textField.pickingMode = PickingMode.Position;
+            uiTransformComponent.Transform.Add(textField);
+            var uiInputComponent = new UIInputComponent();
+            uiInputComponent.TextField = textField;
+            uiInputComponent.Placeholder = new TextFieldPlaceholder(textField);
```

Comment:
I propose you integrate `TextFieldPlaceholder` in the pooling of `TextField` type as from the perspective of our logic they can't be disconnected. Thus, you will avoid the creation of a class every time and will have the placeholder poolable automatically

Reply:
Good idea! Done here: https://github.com/decentraland/unity-explorer/pull/259/commits/d654807d5a0501435eaf50bfc063fefacea735ef